src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -59,7 +60,9 @@ internal sealed class ApplicationExecutor(ILogger<ApplicationExecutor> logger,
                                           IOptions<DcpOptions> options,
                                           IDashboardEndpointProvider dashboardEndpointProvider,
                                           IDashboardAvailability dashboardAvailability,
-                                          DistributedApplicationExecutionContext executionContext)
+                                          DistributedApplicationExecutionContext executionContext,
```

Comment:
These changes look more complex than they are. The main changes are moving `Create{Resource}Async` out to a single method so that multiple resources can be started allocated in parallel. We also publish status updates for starting and failure (if it fails to start).